enterprise/server/remote_execution/containers/docker/docker.go

```diff
@@ -700,8 +700,10 @@ func (r *dockerCommandContainer) Pause(ctx context.Context) error {
 
 func (r *dockerCommandContainer) Remove(ctx context.Context) error {
 	r.removed = true
-	if err := r.client.ContainerRemove(ctx, r.id, dockercontainer.RemoveOptions{Force: true}); err != nil {
-		return wrapDockerErr(err, fmt.Sprintf("failed to remove docker container %s", r.id))
+	if r.id != "" {
```

Comment:
I quickly scanned the code to see that `r.id` is only set if a call to `r.client.ContainerCreate` succeeds.  👍